include/dd/ComplexTable.hpp

```diff
@@ -166,17 +175,20 @@ namespace dd {
             // search in (potentially) lower bucket
             if (lowerKey != key) {
                 ++lower_neighbors;
-                Entry* p_lower = find(table[lowerKey], val);
-                if (p_lower != nullptr) {
+                // buckets are sorted so we only have to look into the last entry of the lower bucket
+                Entry* p_lower = last_entry_table[lowerKey];
+                if (p_lower != nullptr && val - p_lower->value < TOLERANCE) {
                     return p_lower;
                 }
             }
 
             // search in (potentially) higher bucket
             if (upperKey != key) {
                 ++upper_neighbors;
-                Entry* p_upper = find(table[upperKey], val);
-                if (p_upper != nullptr) {
+                // buckets are sorted, we only have to look at the first element
+
+                Entry* p_upper = table[upperKey];
+                if (p_upper != nullptr && p_upper->value - val < TOLERANCE) {
```

Comment:
same as above

Reply:
Again, we can guarentee `p_upper->value > val` :)